drainer/sync/mysql.go

```diff
@@ -38,14 +38,14 @@ type MysqlSyncer struct {
 var createDB = loader.CreateDBWithSQLMode
 
 // NewMysqlSyncer returns a instance of MysqlSyncer
-func NewMysqlSyncer(cfg *DBConfig, tableInfoGetter translator.TableInfoGetter, worker int, batchSize int, queryHistogramVec *prometheus.HistogramVec, sqlMode *string) (*MysqlSyncer, error) {
+func NewMysqlSyncer(cfg *DBConfig, tableInfoGetter translator.TableInfoGetter, worker int, batchSize int, queryHistogramVec *prometheus.HistogramVec, sqlMode *string, destDBType string) (*MysqlSyncer, error) {
 	db, err := createDB(cfg.User, cfg.Password, cfg.Host, cfg.Port, sqlMode)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 
 	var opts []loader.Option
-	opts = append(opts, loader.WorkerCount(worker), loader.BatchSize(batchSize))
+	opts = append(opts, loader.WorkerCount(worker), loader.BatchSize(batchSize), loader.DownstreamTp(destDBType))
```

Comment:
is definition of `DownstreamTp` tricky?  you adjust type of downstream, and only want to use it to check whether we need to keep slave TS.  I suggest that  it's better not to covert it to other type, it only make some people confused

Reply:
updated https://github.com/pingcap/tidb-binlog/pull/670/commits/412a5226bcd11cac6e2422db7c47e3eecc8a6306